packages/pluggable-widgets-tools/src/typings-generator/__tests__/inputs/metadata-attribute.ts

```diff
@@ -0,0 +1,58 @@
+export const attributeMetaDataWebInput = `<?xml version="1.0" encoding="utf-8"?>
+<widget id="mendix.mywidget.MyWidget" needsEntityContext="true" offlineCapable="true"
+    pluginWidget="true" supportedPlatform="Web"
+    xmlns="http://www.mendix.com/widget/1.0/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://www.mendix.com/widget/1.0/ ../xsd/widget.xsd">
+    <properties>
+        <propertyGroup caption="General">
+            <property key="data" type="dataSource">
```

Comment:
Two remarks here:
- This is actually an invalid configuration, as data source should be lower cased and the `isList=true` is required
- We should also test the `isLinked` datasource, we are not exporting that to the client so there shouldn't be a property typing generated for a linked data source

Reply:
Thanks, I should have picked up on the data source getting assigned an `any` type. I also updated the generator to skip linked data sources and added a test case to the datasource test.

Reply:
✨